src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -657,134 +657,180 @@ public void getDocIds(DocIdPusher pusher) throws InterruptedException {
     if (!this.markAllDocsAsPublic) {
       MemberService memberService =
           this.soapFactory.newMemberService(documentManagement);
-      Map<GroupPrincipal, List<Principal>> groupDefinitions =
-          getGroups(memberService);
-      if (this.publicAccessGroupEnabled) {
-        List<Principal> publicAccessGroup = getPublicAccessGroup(memberService);
-        if (publicAccessGroup.size() > 0) {
-          groupDefinitions.put(
-              new GroupPrincipal("[Public Access]", this.localNamespace),
-              publicAccessGroup);
+      DocIdPusher.FeedType feedType = DocIdPusher.FeedType.FULL;
+      SOAPFaultException groupsException = null;
+      Groups groups = new Groups(memberService, this);
+      try {
+        groups.addGroups();
```

Comment:
Does it ever make sense to call addGroups twice on same instance?

Reply:
No, I don't think so. I think as I was figuring out how to get two pieces of information back where we used to only need one, and deciding to wrap the two previous methods in a class, I thought it would be a simpler change to leave the two previous methods alone as much as possible.  Keeping addGroups as a separate method instead of making it the constructor lets the caller check for partial results if an exception was thrown part way through building the groups data, which is what I thought the full/incremental distinction was for.

Reply:
Agreed. Thanks.